app/src/main/java/org/xbmc/kore/ui/ArtistOverviewFragment.java

```diff
@@ -0,0 +1,484 @@
+                                        AlbumListQuery.PROJECTION, null, null, null);
+            case LOADER_SONGS:
+                uri = MediaContract.Songs.buildArtistSongsListUri(hostInfo.getId(), artistId);
+                return new CursorLoader(getActivity(), uri,
+                                        AlbumSongsListQuery.PROJECTION, null, null, AlbumSongsListQuery.SORT);
+            default:
+                return null;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onLoadFinished(Loader<Cursor> cursorLoader, Cursor cursor) {
+        if (cursor != null && cursor.getCount() > 0) {
+            switch (cursorLoader.getId()) {
+                case LOADER_ARTIST:
+                    displayArtistDetails(cursor);
+                    break;
+                case LOADER_ALBUMS:
+                    createAlbumList(cursor);
+                    getLoaderManager().initLoader(LOADER_SONGS, null, this);
+                    break;
+                case LOADER_SONGS:
+                    downloadSongs(cursor);
+            }
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onLoaderReset(Loader<Cursor> cursorLoader) {
+        // Release loader's data
+    }
+
+    @OnClick(R.id.fab)
+    public void onFabClicked(View v) {
+        PlaylistType.Item item = new PlaylistType.Item();
+        item.artistid = artistId;
+        Player.Open action = new Player.Open(item);
+        action.execute(hostManager.getConnection(), new ApiCallback<String>() {
+            @Override
+            public void onSuccess(String result) {
+                if (!isAdded()) return;
+                // Check whether we should switch to the remote
+                boolean switchToRemote = PreferenceManager
+                        .getDefaultSharedPreferences(getActivity())
+                        .getBoolean(Settings.KEY_PREF_SWITCH_TO_REMOTE_AFTER_MEDIA_START,
+                                    Settings.DEFAULT_PREF_SWITCH_TO_REMOTE_AFTER_MEDIA_START);
+                if (switchToRemote) {
+                    int cx = (fabButton.getLeft() + fabButton.getRight()) / 2;
+                    int cy = (fabButton.getTop() + fabButton.getBottom()) / 2;
+                    UIUtils.switchToRemoteWithAnimation(getActivity(), cx, cy, exitTransitionView);
+                }
+            }
+
+            @Override
+            public void onError(int errorCode, String description) {
+                if (!isAdded()) return;
+                // Got an error, show toast
+                Toast.makeText(getActivity(), R.string.unable_to_connect_to_xbmc, Toast.LENGTH_SHORT)
+                     .show();
+            }
+        }, callbackHandler);
+    }
+
+    @OnClick(R.id.add_to_playlist)
+    public void onAddToPlaylistClicked(View v) {
+        final PlaylistType.Item playListItem = new PlaylistType.Item();
+        playListItem.artistid = artistId;
+        MediaPlayerUtils.queueAudio(this, playListItem);
```

Comment:
Breaks the build. You changed the signature of this method in your last PR, so this should reflect that.